gossip/src/cluster_info.rs

```diff
@@ -1972,6 +1972,10 @@ impl ClusterInfo {
     // Returns a predicate checking if the pull request is from a valid
     // address, and if the address have responded to a ping request. Also
     // appends ping packets for the addresses which need to be (re)verified.
+    //
+    // allow lint false positive trait bound requirement (`CryptoRng` only
+    // implemented on `&'a mut T`
+    #[rustversion::attr(since(1.73), allow(clippy::needless_pass_by_ref_mut))]
```

Comment:
This doesn't seems like a false positive to me. The `rng` param seems wrong, or at least very weird, here.

It doesn't need to be `mut rng: &'a mut R,` it should be `rng: &'a mut R` and just passed as `rng` to Ping::new_rand(rng,...)`

Currently we're passing a `&mut &mut R` to `Ping::new_rand` which is probably immediately deref'd, but because we have `&mut rng` it's forcing us to have `mut rng`?

Reply:
Nevermind, I pulled the branch and even with those changes it still flags it falsely.